src/ServicePulse.Host/vue/src/components/failedmessages/MessageView.vue

```diff
@@ -71,51 +76,64 @@ async function getConfiguration() {
 }
 
 async function getEditAndRetryConfig() {
-  const response = await useFetchFromServiceControl("edit/config");
-  const data = await response.json();
-  configuration.value.edit = data;
+  const [, data] = await useTypedFetchFromServiceControl<EditAndRetryConfig>("edit/config");
+
+  editAndRetryConfiguration.value = data;
   return;
 }
 
 function updateMessageDeleteDate() {
-  const countdown = moment(failedMessage.value.last_modified).add(failedMessage.value.error_retention_period, "hours");
-  failedMessage.value.delete_soon = countdown < moment();
-  failedMessage.value.deleted_in = countdown.format();
+  const countdown = moment(failedMessage.value?.last_modified).add(failedMessage.value?.error_retention_period, "hours");
+  if (failedMessage.value) {
+    failedMessage.value.delete_soon = countdown < moment();
+    failedMessage.value.deleted_in = countdown.format();
+  }
 }
 
 async function archiveMessage() {
-  useShowToast("info", "Info", `Deleting the message ${id.value} ...`);
+  useShowToast(TYPE.INFO, "Info", `Deleting the message ${id.value} ...`);
   changeRefreshInterval(1000); // We've started an archive, so increase the polling frequency
-  await useArchiveMessage([id.value]);
-  failedMessage.value.archiving = true;
+  await useArchiveMessage([id.value as string]);
+  if (failedMessage.value) {
+    failedMessage.value.archiving = true;
+  }
 }
 
 async function unarchiveMessage() {
   changeRefreshInterval(1000); // We've started an unarchive, so increase the polling frequency
-  await useUnarchiveMessage([id.value]);
-  failedMessage.value.restoring = true;
+  await useUnarchiveMessage([id.value as string]);
```

Comment:
can id be made a string ref so that the casts aren't required at all usage points?